### PR TITLE
(cheevos) add option to play custom sound on achievement unlock

### PR DESCRIPTION
**Pull request by Jamiras**
## Description

Adds  the ability to play a sound file (`assets/sounds/unlock.*`) when an achievement unlocks.

Builds on #8106, but doesn't require Menu Sounds to be enabled.

General question: How do we provide a default sound file so users won't complain about it not working when they enable it?

## Related Issues

#10503

## Related Pull Requests

n/a

## Reviewers

@twinaphex @meleu @Sanaki

### CODE REVIEW

File: retroarch.c [hunk 1]
```diff
@@ -1141,7 +1141,7 @@ static const camera_driver_t *camera_drivers[] = {
 
 #define AUDIO_BUFFER_FREE_SAMPLES_COUNT (8 * 1024)
 
-#define MENU_SOUND_FORMATS "ogg|mod|xm|s3m|mp3|flac"
+#define MENU_SOUND_FORMATS "ogg|mod|xm|s3m|mp3|flac|wav"
```
Comment (Jamiras) [inline retroarch.c:1144]: Is there a specific reason `wav` was not included in this list? I did notice that the rwav library didn't like any of my `.wav` files that contained metadata.

Comment (Sanaki) [reply retroarch.c:1144]: I can confirm the wav file won't play if any metadata is present. I wouldn't be surprised if that's why it was excluded. Easy to miss that being the cause in testing.

**Comment by meleu** [comment]
Built and tested here, and **hey!** That's really cool! :smile: 
Working as expected.

**Comment by hizzlekizzle** [comment]
I think people will be very excited for this.

**Pull request merged**